risc0/bigint2/src/rsa/mod.rs

```diff
@@ -45,12 +45,16 @@ fn to_u32_digits(input: &BigUint) -> Vec<u32> {
         digits.push(word);
     }
 
+    digits.resize(RSA_3072_WIDTH_WORDS, 0);
```

Comment:
you're completely right, and it should be a panic for larger values. Will have a test case for the larger bits and make sure the final impl handles it correctly

Reply:
yes, that is correct that the `sys_bigint2_3` allows smaller buffers, but I wonder if that is intentional to not be strict?

That is also the most efficient way in terms of cycle count, so it would be preferred if there is no security concern (I can't think of one, but haven't dug into internals of the syscall yet) [`4c0e739` (#2510)](https://github.com/risc0/risc0/pull/2510/commits/4c0e739cb6b851e160d4ba13ef2571dc25277f0a)

Reply:
Ah, sorry, I didn't mean adjust the existing assert to be weaker, I meant adding a new assert before resizing so that you would only ever resize up

Reply:
The interpreter on the host side will read values that it's not intended to, which is a bug but not a soundness issue. However the verifier circuitry will read incorrect values which is will be a soundness problem.

Reply:
We also need to avoid a resize entirely.

Reply:
gotcha, sorry I misread last night, and I assumed the host function was implemented passing a buffer length so that there was never a read past the buffer. Will fix now